src/module/hotKeysModule.js

```diff
@@ -251,12 +276,32 @@ var HotKeysModule = class HotKeysModule {
             this.actionNameToActionMap.set(KeyBindingAction[actionKey], () => {
                 const currentNumOfWorkspaces =
                     Me.msWorkspaceManager.msWorkspaceList.length - 1;
+                const currentWorkspaceIndex =
+                    this.workspaceManager.get_active_workspace_index();
+                let nextWorkspaceIndex = workspaceIndex;
 
-                // go to new workspace if attemping to go to index bigger than currently available
-                Me.msWorkspaceManager.primaryMsWorkspaces[
-                    workspaceIndex > currentNumOfWorkspaces
+                if (
+                    this.lastWorkspaceIndex === null ||
+                    (nextWorkspaceIndex !== this.lastNextWorkspaceIndex)
+                    ) {
+                    this.lastWorkspaceIndex = currentWorkspaceIndex;
+                    this.lastNextWorkspaceIndex = nextWorkspaceIndex;
+                } else {
+                    if (nextWorkspaceIndex === this.lastNextWorkspaceIndex) {
+                        nextWorkspaceIndex = this.lastWorkspaceIndex;
+                    }
+                    this.lastWorkspaceIndex = null;
+                    this.lastNextWorkspaceIndex = null;
+                }
+
+                // go to new workspace if attempting to go to index bigger than currently available
+                nextWorkspaceIndex =
+                    nextWorkspaceIndex > currentNumOfWorkspaces
                         ? currentNumOfWorkspaces
-                        : workspaceIndex
+                        : nextWorkspaceIndex;
+
+                Me.msWorkspaceManager.primaryMsWorkspaces[
+                    nextWorkspaceIndex
                 ].activate();
```

Comment:
First `this.lastNextWorkspaceIndex` seem to always equal to `this.workspaceManager.get_active_workspace_index()`.

Then I think it's can produce side effect. Since you save previous index in this function only that's mean that when you using other way to navigate around workspaces well it's not accurate anymore.

For example:
- `I'm workspace 4`
- `I press Super+2 and navigate to Workspace 2`
- `I press Super+W and navigate to Workspace 1`
- `I press Super+2 to go back to workspace 2 and It's move me to workspace 4`

One way to get around this would be to subscribe to the `switch-workspace` to save the previous workspace all the time and when navigating by `Super+{NUMBER}` check if `currentWorkspaceIndex === {NUMBER} ` and if so go back to the previous workspace but even in that case for me it's would have a undesired behavior because when you press `Super+{Number}` the only thing you should want and expect is to navigate to the given workspace.

So to restrict the `magic shortcut` to the only **use case** we want we should stash `PreviousWorkspaceIndex` only after navigating from `Super+{NUMBER}` and subscribe to the `switch-workspace` to clear this value if we navigate elsewhere.

Reply:
> First `this.lastNextWorkspaceIndex` seem to always equal to `this.workspaceManager.get_active_workspace_index()`.

No, it is never this value. I changed the names, hopefully it is clearer now. I also fixed the bug when using other ways to activate another workspace, thanks for the insights.